Error while loading ONNX model

### System Information

I've trained yolov10 model and exported it in .onnx. Then I want to use it in my c++ code, but getting error during processing node with 2 inputs and 2 outputs.
OpenCV version: 4.13.0 Operating System / Platform: Windows 11

### Detailed description

Here is full output with this error:
```
[ERROR:0@0.126] global onnx_importer.cpp:1040 cv::dnn::dnn4_v20251223::ONNXImporter::handleNode DNN/ONNX: ERROR during processing node with 2 inputs and 2 outputs: [TopK]:(onnx_node!/model.23/TopK_1) from domain='ai.onnx'
OpenCV: terminate handler is called! The last OpenCV error is:
OpenCV(4.13.0) Error: Unspecified error (> Node [TopK@ai.onnx]:(onnx_node!/model.23/TopK_1) parse error: OpenCV(4.13.0) C:\Users\pankovn\Downloads\opencv-4.13.0\modules\dnn\src\layers\topk_layer.cpp:102: error: (-2:Unspecified error) in function 'bool __cdecl cv::dnn::TopKLayerImpl::getMemoryShapes(const class std::vector<class std::vector<int,class std::allocator<int> >,class std::allocator<class std::vector<int,class std::allocator<int> > > > &,const int,class std::vector<class std::vector<int,class std::allocator<int> >,class std::allocator<class std::vector<int,class std::allocator<int> > > > &,class std::vector<class std::vector<int,class std::allocator<int> >,class std::allocator<class std::vector<int,class std::allocator<int> > > > &) const'
> > TopK: K is out of range (expected: 'K < input_shape[axis_normalized]'), where
> >     'K' is 300
> > must be less than
> >     'input_shape[axis_normalized]' is 300
> ) in cv::dnn::dnn4_v20251223::ONNXImporter::handleNode
```

### Steps to reproduce

```cpp
#include <opencv2/core/utils/filesystem.hpp>
#include <opencv2/highgui/highgui.hpp>
#include <opencv2/imgproc/imgproc.hpp>

int main(){
    cv::dnn::Net m_net;
    m_net = cv::dnn::readNetFromONNX("models/my_model.onnx");
}

```

### Issue submission checklist

- [x] I report the issue, it's not a question
- [x] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [x] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/dnn/src/layers/topk_layer.cpp
+++ b/modules/dnn/src/layers/topk_layer.cpp
@@ -97,9 +97,9 @@ public:
         // Normalize axis
         int axis_normalized = normalize_axis(axis, input_shape.size());
 
-        // Check if K is in range (0, input_shape[axis])
+        // Check if K is in range (0, input_shape[axis]]
         CV_CheckGT(K, 0, "TopK: K needs to be a positive integer");
-        CV_CheckLT(K, input_shape[axis_normalized], "TopK: K is out of range");
+        CV_CheckLE(K, input_shape[axis_normalized], "TopK: K is out of range");
 
         // Assign output shape
         auto output_shape = input_shape;

